graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphQLServiceBuilder.java

```diff
@@ -0,0 +1,258 @@
+import graphql.execution.instrumentation.ChainedInstrumentation;
+import graphql.execution.instrumentation.Instrumentation;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.GraphQLTypeVisitor;
+import graphql.schema.SchemaTransformer;
+import graphql.schema.idl.RuntimeWiring;
+import graphql.schema.idl.RuntimeWiring.Builder;
+import graphql.schema.idl.SchemaGenerator;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+
+/**
+ * Constructs a {@link GraphQLService} to serve GraphQL within Armeria.
+ */
+@UnstableApi
+public final class GraphQLServiceBuilder {
+
+    private static final Logger logger = LoggerFactory.getLogger(GraphQLServiceBuilder.class);
+
+    private static final ImmutableList<String> DEFAULT_SCHEMA_FILE_NAMES = ImmutableList.of("schema.graphqls",
+                                                                                            "schema.graphql");
+    private final ImmutableList.Builder<File> schemaFileBuilder = ImmutableList.builder();
+
+    private final ImmutableList.Builder<RuntimeWiringConfigurator> runtimeWiringConfiguratorBuilder =
+            ImmutableList.builder();
+    private final ImmutableList.Builder<Consumer<DataLoaderRegistry>> dataLoaderRegistryConsumerBuilder =
+            ImmutableList.builder();
+    private final ImmutableList.Builder<GraphQLTypeVisitor> typeVisitorBuilder = ImmutableList.builder();
+    private final ImmutableList.Builder<Instrumentation> instrumentationBuilder = ImmutableList.builder();
+    private final ImmutableList.Builder<Consumer<GraphQL.Builder>> graphQLBuilderConsumerBuilder =
+            ImmutableList.builder();
+
+    private boolean useBlockingTaskExecutor;
+
+    GraphQLServiceBuilder() {}
+
+    /**
+     * Adds the schema {@link File}s.
+     * If not set, the {@code schema.graphql} or {@code schema.graphqls} will be imported from the resource.
+     */
+    public GraphQLServiceBuilder schemaFile(File... schemaFiles) {
+        return schemaFile(ImmutableList.copyOf(requireNonNull(schemaFiles, "schemaFiles")));
+    }
+
+    /**
+     * Adds the schema {@link File}s.
+     * If not set, the {@code schema.graphql} or {@code schema.graphqls} will be imported from the resource.
+     */
+    public GraphQLServiceBuilder schemaFile(Iterable<? extends File> schemaFiles) {
+        schemaFileBuilder.addAll(requireNonNull(schemaFiles, "schemaFiles"));
+        return this;
+    }
+
+    /**
+     * Adds the {@link DataLoaderRegistry} consumers.
+     */
+    public GraphQLServiceBuilder configureDataLoaderRegistry(Consumer<DataLoaderRegistry>... configurers) {
+        requireNonNull(configurers, "configurers");
+        return configureDataLoaderRegistry(ImmutableList.copyOf(configurers));
+    }
+
+    /**
+     * Adds the {@link DataLoaderRegistry} consumers.
+     */
+    public GraphQLServiceBuilder configureDataLoaderRegistry(
+            Iterable<Consumer<DataLoaderRegistry>> configurers) {
```

Comment:
It looks weird but. 
```suggestion
            Iterable<? extends Consumer<? super DataLoaderRegistry>> configurers) {
```
https://github.com/line/armeria/blob/e872d75a03de3bc384b7416d16c82dfd73187ccb/core/src/main/java/com/linecorp/armeria/server/ServerListenerBuilder.java#L191